File: Sohi.Models/Plan.cs
using System;
namespace Sohi.Models
{
    public class Plan
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public string Type { get; set; }

        public Decimal Price { get; set; }

        public string BillingPeriod { get; set; }

        public string ProductId { get; set; }

        public Decimal Tax { get; set; }

        public Decimal Total { get; set; }

        public DateTime CreatedOn { get; set; }

        public DateTime ModifiedOn { get; set; }

    }
}

File: backend/app/models/plan.py
"""Subscription Plan. Ports C# `Plan` (money fields keep Numeric(18,2) per EF config)."""
from datetime import datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID, uuid4

from sqlalchemy import Column, Numeric
from sqlmodel import Field, SQLModel

_MONEY = lambda: Column(Numeric(18, 2), nullable=False)  # noqa: E731


class Plan(SQLModel, table=True):
    __tablename__ = "plans"

    id: UUID = Field(default_factory=uuid4, primary_key=True)
    name: Optional[str] = None
    type: Optional[str] = None
    price: Decimal = Field(default=Decimal("0"), sa_column=_MONEY())
    billing_period: Optional[str] = None
    product_id: Optional[str] = None  # Stripe product/price id
    tax: Decimal = Field(default=Decimal("0"), sa_column=_MONEY())
    total: Decimal = Field(default=Decimal("0"), sa_column=_MONEY())
    created_on: Optional[datetime] = Field(default_factory=datetime.utcnow)
    modified_on: Optional[datetime] = None
